### ISSUE TITLE
Roberta Tokenizer cannot handle inputs with `<mask>` token

### ISSUE THREAD

**Issue by liyucheng09** [huggingface/transformers#11415]
## Environment info

- `transformers` version: 4.5.1
- Platform: macOS-11.1-arm64-arm-64bit
- Python version: 3.9.1
- PyTorch version (GPU?): 1.8.0 (False)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

## Description

It'a bizarre bug. When I encode a string with a `<mask>` token and decode it immediately, the space in front of the `<mask>` token disappears. 

```
>>> from transformers import AutoTokenizer
>>> tokenizer=AutoTokenizer.from_pretrained('roberta-base')
>>> tokenized_inputs=tokenizer('I <mask> you')['input_ids']
>>> tokenizer.decode(tokenized_inputs)
'<s>I<mask> you</s>'
>>> 
```

It will lead to many problems. Such as `pipeline('fill-mask')` cannot provides valid results.

```
>>> from transformers import pipeline
>>> nlp=pipeline('fill-mask')
>>> nlp.tokenizer
PreTrainedTokenizerFast(name_or_path='distilroberta-base', vocab_size=50265, model_max_len=512, is_fast=True, padding_side='right', special_tokens={'bos_token': '<s>', 'eos_token': '</s>', 'unk_token': '<unk>', 'sep_token': '</s>', 'pad_token': '<pad>', 'cls_token': '<s>', 'mask_token': AddedToken("<mask>", rstrip=False, lstrip=True, single_word=False, normalized=False)})
>>> nlp('app <mask>')
{'score': 0.09366267919540405,
 'sequence': 'appeal',
 'token': 18696,
 'token_str': 'eal'}
```

It seems that the mask-filling process omits the space, which is not what we expected (we expect the token to be filled in mask is another word rather than a sub-word, since there is a space as a separator.) 

Does anyone notice the same issue?

**Comment by LysandreJik** [comment]
Hi, the tokenizer of RoBERTa is a byte-level BPE tokenizer. It is a subword tokenizer.

In your example with `app ` it completes the word, but there also exists full words within it's dictionary, as you can see when the input sequence is appropriate:

```py
>>> nlp('Hello, how are you <mask> sir?')
[
    {'sequence': 'Hello, how are you doing sir?', 'score': 0.6784416437149048, 'token': 608, 'token_str': ' doing'}, 
    {'sequence': 'Hello, how are you feeling sir?', 'score': 0.08236288279294968, 'token': 2157, 'token_str': ' feeling'}, 
    {'sequence': 'Hello, how are you, sir?', 'score': 0.06469670683145523, 'token': 6, 'token_str': ','}, 
    {'sequence': 'Hello, how are you looking sir?', 'score': 0.04527667537331581, 'token': 546, 'token_str': ' looking'}, 
    {'sequence': 'Hello, how are you going sir?', 'score': 0.02970985323190689, 'token': 164, 'token_str': ' going'}]
```
